mvnw

```diff
@@ -105,6 +106,21 @@ trim() {
   printf "%s" "${1}" | tr -d '[:space:]'
 }
 
+# find the Python executable
+if [ -z "${PYTHON_CMD-}" ]; then
+  for PYTHON_PATH in /usr/bin/python3 /usr/local/bin/python3 /usr/bin/python /usr/local/bin/python; do
+    if [ -x "${PYTHON_PATH}" ]; then
+      export PYTHON_CMD="${PYTHON_PATH}"
+      break
+    fi
+  done
+
+  if [ -z "${PYTHON_CMD-}" ]; then
+    echo "Python executable not found; please set PYTHON_CMD" >&2
+    return 1
+  fi
+fi
```

Comment:
fixed in 7af4f4ce001d5f3c6fb22b208c10535d109d3fc3

Reply:
Thanks for the adjusted approach @fgerlits.

Reviewing the test itself, it seems like this is more suitable as an integration test under the `nifi-system-test-suite` module. The `system-tests` GitHub workflow already installs Python, and it is a better place to run longer types of tests, such as this one that require installing dependency sets. What do you think about refactoring this test into the system test module?

Reply:
You are right, a unit test should not download stuff from the internet (and take 20+ seconds).  In e3ab1c7c53742fe22cddcc8384018e3a55e93f81, I have mocked out the `subprocess.run` call, and now only test that it gets called once instead of twice, as it used to be.

I also agree it would be a good idea to add an integration test which does the package installations and version checks; can we do that separately, in another pull request?

Reply:
This unit test has value: it loads the ExtensionManager class, so if there are e.g. syntax errors, they get caught earlier; and it validates the change in the PR in the sense that if you run the test with the ExtensionManager code on `main`, it will fail: we are now calling `pip install` once instead of twice.

I have renamed and refactored it to make it clearer that this is just a unit test for a single method in the ExtensionManager class, and also to make it easier to add unit tests for the rest of the methods later.